datahost-ld-openapi/src/tpximpact/datahost/ldapi/util/data/validation.clj

```diff
@@ -234,11 +234,15 @@
 (defn dataset-ctor-opts [value]
   (-dataset-ctor-opts value))
 
-(defn- parse-double*
+(defn parse-double*
   "Returns a double or nil."
   [v]
-  (when (re-find #"^\-{0,1}\d+\.\d+$" v)
-    (clojure.core/parse-double v)))
+  (case v
+    "INF" Double/POSITIVE_INFINITY
+    "-INF" Double/NEGATIVE_INFINITY
+    "+INF" Double/POSITIVE_INFINITY
+    (when (re-find #"(\+|-)?([0-9]+(\.[0-9]*)?|\.[0-9]+)([Ee](\+|-)?[0-9]+)?|(\+|-)?INF|NaN" v)
+      (clojure.core/parse-double v))))
```

Comment:
If we need to parse XSD conforming strings, then maybe we could just   `javax.xml.bind.DatatypeConverter/parseDouble`?

Reply:
Thanks @rosado, I agree that's a much better idea.

Though I've just tried and it seems that method doesn't actually support the literal `+INF`, not that I'd expect to see that very often, but it is [explicitly listed in the spec](https://www.w3.org/TR/xmlschema11-2/#sec-lex-double) as one of the four literal lexical values, so it does make me question the spec compliance of that class.

I'll do some more looking to see what RDF4j, for example, does under the hood 👀

Reply:
It looks like RDF4j has the same issue, [I've filled it here](https://github.com/eclipse-rdf4j/rdf4j/issues/4881) though this isn't a big deal.

I think we should probably use the RDF4j class as the maintainers have historically been very responsive around parsing issues etc, and it also supports a wider set of XSD data types which I think we'd ultimately like to support